src/drt/src/io/io_parser_helper.cpp

```diff
@@ -50,116 +50,125 @@ void io::Parser::initDefaultVias()
+          DRT, 608, "Could not find user defined via {}", userDefinedVia);
+    }
+    auto viaDef = tech->getVia(userDefinedVia);
+    tech->getLayer(viaDef->getCutLayerNum())->setDefaultViaDef(viaDef);
+  }
   std::map<frLayerNum, std::map<int, std::map<viaRawPriorityTuple, frViaDef*>>>
       layerNum2ViaDefs;
   for (auto layerNum = design->getTech()->getBottomLayerNum();
        layerNum <= design->getTech()->getTopLayerNum();
        ++layerNum) {
-    if (design->getTech()->getLayer(layerNum)->getType()
-        != dbTechLayerType::CUT) {
+    auto layer = design->getTech()->getLayer(layerNum);
+    if (layer->getType() != dbTechLayerType::CUT) {
       continue;
     }
-    for (auto& viaDef : design->getTech()->getLayer(layerNum)->getViaDefs()) {
-      int cutNum = int(viaDef->getCutFigs().size());
-      viaRawPriorityTuple priority;
-      getViaRawPriority(viaDef, priority);
-      layerNum2ViaDefs[layerNum][cutNum][priority] = viaDef;
-    }
-    if (!layerNum2ViaDefs[layerNum][1].empty()) {
-      auto defaultSingleCutVia
-          = (layerNum2ViaDefs[layerNum][1].begin())->second;
-      tech->getLayer(layerNum)->setDefaultViaDef(defaultSingleCutVia);
-    } else {
-      if (layerNum >= BOTTOM_ROUTING_LAYER) {
-        logger->error(DRT,
-                      234,
-                      "{} does not have single-cut via.",
-                      tech->getLayer(layerNum)->getName());
-      }
-    }
-    // generate via if default via enclosure is not along pref dir
-    if (ENABLE_VIA_GEN && layerNum >= BOTTOM_ROUTING_LAYER) {
-      auto techDefautlViaDef = tech->getLayer(layerNum)->getDefaultViaDef();
-      frVia via(techDefautlViaDef);
-      Rect layer1Box;
-      Rect layer2Box;
-      frLayerNum layer1Num;
-      frLayerNum layer2Num;
-      via.getLayer1BBox(layer1Box);
-      via.getLayer2BBox(layer2Box);
-      layer1Num = techDefautlViaDef->getLayer1Num();
-      layer2Num = techDefautlViaDef->getLayer2Num();
-      bool isLayer1Square = (layer1Box.xMax() - layer1Box.xMin())
-                            == (layer1Box.yMax() - layer1Box.yMin());
-      bool isLayer2Square = (layer2Box.xMax() - layer2Box.xMin())
-                            == (layer2Box.yMax() - layer2Box.yMin());
-      bool isLayer1EncHorz = (layer1Box.xMax() - layer1Box.xMin())
-                             > (layer1Box.yMax() - layer1Box.yMin());
-      bool isLayer2EncHorz = (layer2Box.xMax() - layer2Box.xMin())
-                             > (layer2Box.yMax() - layer2Box.yMin());
-      bool isLayer1Horz
-          = (tech->getLayer(layer1Num)->getDir() == dbTechLayerDir::HORIZONTAL);
-      bool isLayer2Horz
-          = (tech->getLayer(layer2Num)->getDir() == dbTechLayerDir::HORIZONTAL);
-      bool needViaGen = false;
-      if ((!isLayer1Square && (isLayer1EncHorz != isLayer1Horz))
-          || (!isLayer2Square && (isLayer2EncHorz != isLayer2Horz))) {
-        needViaGen = true;
+    if (layer->getDefaultViaDef() == nullptr) {
```

Comment:
if (layer->getDefaultViaDef()) {
   continue;
}

seems equivalent and clearer than the huge if scope.

Reply:
I normally do this but this week I was reading docs/contrib/CodingPractices and practice 9 states otherwise. Honestly I hate it that most of the bad coding practices examples are from TritonRoute :smile: 
I changed it anyways